tests/core/engine_adapter/test_integration.py

```diff
@@ -376,6 +395,116 @@ def non_temp_tables(self) -> t.List[str]:
         return [x for x in self.tables if not x.startswith("__temp") and not x.startswith("temp")]
 
 
+class ProjectCreator(PydanticModel):
```

Comment:
I had trouble creating a test case to reproduce the issue because it touched so many concepts.

What I really wanted was to be able to set up a small isolated SQLMesh project to expose the issue from the users perspective (`sqlmesh plan` throws error when I use a model defined like so), so I wrote this small fixture to create a minimal project I could run against the current engine adapter

I realise there is some overlap with `sqlmesh.cli.example_project.init_example_project` but I didnt need the example models and they just slowed things down